reconcile/test/test_ocm_upgrade_scheduler.py

```diff
@@ -208,6 +209,89 @@ def test_calculate_diff_inter_lock(self, ocm_map):
         assert diffs == expected
 
 
+class TestUpgradeableVersion:
+
+    @staticmethod
+    @pytest.fixture
+    @patch("reconcile.ocm_upgrade_scheduler.OCMMap", autospec=True)
+    def ocm(mock_ocm_map):
+        map = mock_ocm_map.return_value
+        ocm = map.get.return_value
+        ocm.get_available_upgrades.return_value = ["4.3.5", "4.3.6", "4.4.1"]
+        ocm.version_blocked.return_value = False
+        return map.get("foo")
+
+    @staticmethod
+    @pytest.fixture
+    def ocm_gated(ocm):
+        ocm.get_version_gates.side_effect = lambda v: [{"id": 1, "version_raw_prefix": 4.4}] if v == "4.4" else []
+        return ocm
+
+    @staticmethod
+    @pytest.fixture
+    def upgrade_policy() -> dict[str, any]:
+        return {
+            "current_version": "4.3.5",
+            "channel": "stable",
+            "workloads": "test",
+            "conditions": {
+                "soakdays": 1,
+            },
+        }
+
+    def test_upgradeable_version_blocked(self, upgrade_policy, ocm):
+        ocm.version_blocked.return_value = True
+        x = ous.upgradeable_version(upgrade_policy, {}, ocm)
```

Comment:
nit: would be clearer when replacing `x = ...` by `version, needs_agreement = ...`

Reply:
replaced by single return value in subsequent commit.